core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -147,6 +147,7 @@ public String location() {
     return properties;
   }
 
+  // Note that schema parsed from string does not contain the correct schema ID.
```

Comment:
This is related to the comment I had in https://github.com/apache/iceberg/pull/2465#discussion_r612661016 that we didn't persist schema id within `toJson()` which is called when we serialize the table. 

Honestly for this case I think we can go either way; we can use a different `toJson` implementation when serialize the table since schema at that time almost guaranteed to be the original schema from table metadata; however since the only usage of `schemaId` is for time travel queries, and this use case doesn't need id from the current `schema` itself so adding it isn't necessary, and not having schema Id is not something we don't expect per [this note](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/Schema.java#L44-L45).

Reply:
I think we should use the correct `toJson` implementation so that we don't need to note that the ID doesn't match.

Reply:
This is actually before we decided to write down schema id as part of `toJson`; now that `toJson` always writes schema id, this note is no longer relevant and I'll remove it.